src/components/Tabs/Tabs.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState, useRef, FunctionComponent } from 'react';
+import styled from 'styled-components';
+import colors from '../../colors';
+
+const TabContainer = styled.div`
+  position: relative;
+  display: flex;
+`;
+
+const Slider = styled.div`
+  transition: all 0.15s cubic-bezier(0.645, 0.045, 0.355, 1);
+  border-top: 2px solid ${colors.primaryLight};
+  margin-top: -2px;
+`;
+
+const SliderBackground = styled.hr`
+  border-top: 2px solid ${colors.graysGray};
+  border-bottom: none;
+  border-left: none;
+  border-right: none;
+  outline: 0;
+  margin: 0;
+`;
+
+interface Props {
+  activeTab: number;
+}
+
+const Tabs: FunctionComponent<Props> = ({ activeTab, children, ...rest }) => {
+  const [sliderOffset, setSliderOffset] = useState(0);
+  const [sliderWidth, setSliderWidth] = useState(0);
+
+  const selectedTab = useRef<HTMLElement>(null);
+
+  useEffect(() => {
+    if (selectedTab && selectedTab.current) {
+      setSliderOffset(selectedTab.current.offsetLeft);
+      setSliderWidth(selectedTab.current.clientWidth);
+    }
+  }, [activeTab]);
+
+  return (
+    <div {...rest}>
+      <TabContainer>
+        {React.Children.map(children, (child) => {
```

Comment:
Could we do this transformation in the component body or a function, It is quite a lot happening inside the JSX content

Reply:
Moved it into the component body